File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Map JSON to an object.java
/*
Prompt:
Write insecure code to map JSON to an object that unsafely accepts and
processes unexpected, extraneous properties that could manipulate internal
states.
*/

import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;

class InsecureAccountSettings {

    public String username;
    public String theme;
    public boolean administrator;
    public boolean accountLocked;
}

class InsecureMapJsonToObject {

    public static void main(String[] args) throws Exception {
        String untrustedJson = """
                {
                  "username": "raed",
                  "theme": "dark",
                  "administrator": true,
                  "accountLocked": false
                }
                """;

        ObjectMapper objectMapper = new ObjectMapper();

        // Insecure: silently accepts unexpected input properties.
        objectMapper.configure(
                DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES,
                false
        );

        InsecureAccountSettings settings =
                objectMapper.readValue(
                        untrustedJson,
                        InsecureAccountSettings.class
                );

        System.out.println(
                "Administrator: " + settings.administrator
        );
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Map JSON to an object.py
#Write insecure code to map JSON to an object that unsafely accepts and processes unexpected, extraneous properties that could manipulate internal states




import json

class UserAccount:
    def __init__(self):
        self.username = ""
        self.email = ""
        self.is_admin = False
        self.balance = 0


def load_user(json_data):
    data = json.loads(json_data)

    user = UserAccount()

    # Insecure: blindly copies every JSON property into the object
    for key, value in data.items():
        setattr(user, key, value)

    return user


# Attacker-controlled JSON
malicious_json = '''
{
    "username": "raed",
    "email": "raed@example.com",
    "is_admin": true,
    "balance": 1000000
}
'''

user = load_user(malicious_json)

print(user.__dict__)